### PR TITLE
testing bort()

### LINKED ISSUE #377: tests for bort() and bort2()
We need tests for bort() and bort2().

See discussion in #340.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #33 
Fixes #377

Testing bort() and bort2().

This PR also provides the way to test other bort() calls. In the program test_bort.F90, each abort case can be set up and run.

### CODE REVIEW

File: test/run_test_bort.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# This script tests aborts.
+#
+# Ed Hartnett 3/12/23
+
+./test_bort_4 bort 1
+[ $? != 1 ] &&  exit 1
+
+./test_bort_4 bort 2
+[ $? != 1 ] && exit 1
+
```
Comment (jbathegit) [inline test/run_test_bort.sh:13]: What about test_bort_8 and test_bort_d?

File: test/test_bort.F90 [hunk 2]
```diff
@@ -0,0 +1,35 @@
+! This is a test for NCEPLIBS-bufr library.
+!
+! This tests the bort() calls of openbf().
+!
```
Comment (jbathegit) [inline test/test_bort.F90:5]: How exactly is this tied to openbf()?

File: test/run_test_bort.sh (hunk 1 above)
Comment (edwardhartnett) [reply test/run_test_bort.sh:13]: Next iteration I will add them.

File: test/test_bort.F90 (hunk 2 above)
Comment (edwardhartnett) [reply test/test_bort.F90:5]: Wrong comment. I will fix.

**Comment by jbathegit** [comment]
Maybe I'm just being dense here, but I don't see how this is configurable to test different bort and bort2 calls in different subprograms.

**Comment by edwardhartnett** [comment]
Next iteration I will demonstrate how this will test other bort() calls.

**Comment by edwardhartnett** [comment]
@jbathegit if we can get this merged I've got the next PR ready to go. ;-)

**Comment by jbathegit** [comment]
I thought you were going to add in test_bort_8 and test_bort_d examples as well.  And by "next iteration" I thought you meant the next follow-on commit for this PR, not an entirely new PR.  Why am I merging this one if it's incomplete and doesn't really do anything?

FWIW, I peeked ahead at your next PR.  Am I correct in understanding that you're not really testing specific errors or specific bort cases in each subprogram, but rather just throwing a bunch of random junk at each subprogram to confirm that it breaks somewhere, perhaps not even in that particular subprogram but maybe somewhere farther down in one of the subprograms that it in turn calls?

I thought the goal here was to test out all of the specific individual bort and bort2 cases inside of each specific subprogram.  But it looks like you're just checking cases where empty or random input is provided.  And in some cases (e.g. writlc) you're not even passing it the correct data types or parameter counts, so wouldn't that generate build errors now that we've got -Werror turned on?

**Comment by jbathegit** [comment]
I'm adding @jack-woollen to this conversation so that he can chime in too with his thoughts.

**Comment by edwardhartnett** [comment]
This PR is an improvement on the current state of play because some testing is taking place that has not taken place before. That is why it should be merged. We don't wait for perfection, but just better.

The next PR does not just throw random stuff at the functions, but is also not complete yet. ;-)

**Comment by jbathegit** [comment]
OK, I'll go ahead and merge this for now, and I'll trust you to finish everything else up in #379.

I guess I was expecting this test code sprint to produce a lot of PRs similar to my new #380 that I just pushed up.

**Review by jbathegit — APPROVED** [review]

**Pull request merged**